guidelines/guidelines_django.md

```diff
@@ -0,0 +1,68 @@
+
+There are many approaches available and you can read about them all around the internet. The one we adopt to our projects is to have a shared services module where all the business logic should be placed. This means that you are encouraged to use Django tools as long as the business logic is in the `services` module. For instance, you should use Django's class based views but instead of writing the [business] logic in the Django class you will write it as a function in the `services` module, import and call the function. Django related code (including code that glues `services` code so it adapts to Django) should be in the Django apps, everything else should go in the `services` module. 
+
+Here are some practical guides on how you should organize services:
+
+- Service directory looks like this:
+
+```
+services/
+├── __init__.py
+├── {service_name}                          # This is a toplevel namespace
+│   ├── __init__.py
+│   ├── {service_module1}.py                # This is the specific namespace for a group of services
+│   ├── {service_module2}.py
+│   └── tests                               # Tests folder is always in the root of the service
+│       ├── __init__.py
+│       ├── test_{service_module1}.py
+│       └── test_{service_module2}.py
+├── {other_service}
+│   ├── __init__.py
+│   ├── ...
+│   └── tests
+│       ├── __init__.py
+│       └── ...
+└── tasks                                   # The `tasks` service is the only place were tasks should live
+    ├── __init__.py
+    ├── {identifier}_tasks.py
+    └── tests
+        ├── __init__.py
+        └── test_{identifier}_tasks.py
+```
+
+- Services are Python code that happen to import from Django `models`. They are not tied to any specific way of writing code defined by the framework. 
+
+- Services never import from a Django app except for `models`.
+
+- No business logic goes inside the a Django app. You import it from a service and call it.
+
+- When using a service you should never import (or chain) more than 3 level down in the module. Eg.: 
+
+```
+from services.integrations import google_calendar  # good
+from services.integrations.google_calendar import get_credentials # bad
+
+google_calendar.get_credentials(user) # good
+google_calendar.credentials.get_credentials(user) # bad
+```
+
+- Exceptions should be available at the root of the service. 
+
+Explanation: it would be misleading to do `from services.service_name import exception` as you might need exceptions from multiple services in the same module leading to the need of using alias. Also we don't want to use exceptions as `service_name.Exception`, for most cases the exception name is enough to identify what it means.
```

Comment:
IMO this first sentence is a bit long. An option would be to break it into 2 sentences.